simplified-services-api/src/main/java/org/librarysimplified/services/api/Services.kt

```diff
@@ -2,30 +2,28 @@ package org.librarysimplified.services.api
 
 import com.google.common.util.concurrent.ListenableFuture
 import com.google.common.util.concurrent.SettableFuture
-import java.lang.IllegalStateException
+import org.slf4j.LoggerFactory
 import java.util.concurrent.TimeUnit
 
 object Services : ServiceDirectoryProviderType {
 
+  private val logger = LoggerFactory.getLogger(Services::class.java)
   private val servicesLock: Any = Any()
   private var servicesDirectory: ServiceDirectoryType? = null
   private val servicesFuture: SettableFuture<ServiceDirectoryType> = SettableFuture.create()
 
   override fun serviceDirectory(): ServiceDirectoryType {
-    return synchronized(this.servicesLock) {
-      this.servicesDirectory ?: throw IllegalStateException("No service directory has been created!")
+    try {
+      return this.servicesFuture.get(3L, TimeUnit.MINUTES)
```

Comment:
Three minutes is a very long time in the world of mobile apps. I'd feel a little more comfortable sticking with 30-60 seconds. Have we ever seen it taking longer than that?

Reply:
Let's go back to 30.